In [10]:
import pandas as pd

In [14]:
#a
df = pd.read_csv("Marks.csv", delimiter=",")
df = df.drop(columns=["Unnamed: 2"])

df_part1 = df.iloc[:, :2]
df_part2 = df.iloc[:, 2:]

df_part1.columns = ["Hours_of_Study", "Score"]
df_part2.columns = ["Hours_of_Study", "Score"]

combined_df = pd.concat([df_part1, df_part2], ignore_index=True)

print(combined_df)

    Hours_of_Study      Score
0                5  52.122100
1                5  52.122100
2                7  72.122100
3                3  32.122100
4                0   2.122104
5                5  52.122100
6                2  22.122100
7                7  72.122100
8                4  42.122100
9                2  22.122100
10               1  12.122100
11               6  62.122100
12               8  82.122100
13               5  52.122100
14               7  72.122100
15               8  82.122100
16               7  72.122100
17               8  82.122100
18               8  82.122100
19               8  82.122100
20               0   2.122104
21               8  82.122100
22               7  72.122100
23               4  42.122100
24               6  62.122100
25               7  72.122100
26               2  22.122100
27               0   2.122104
28               8  82.122100
29               4  42.122100


In [15]:
X=combined_df['Hours_of_Study']
y=combined_df['Score']

In [16]:
import statsmodels.api as sm


model = sm.OLS(y, X).fit()
results_summary = model.summary()

In [17]:
results_summary

<class 'statsmodels.iolib.summary.Summary'>
"""
                                 OLS Regression Results                                
=======================================================================================
Dep. Variable:                  Score   R-squared (uncentered):                   1.000
Model:                            OLS   Adj. R-squared (uncentered):              1.000
Method:                 Least Squares   F-statistic:                          1.037e+05
Date:                Wed, 30 Oct 2024   Prob (F-statistic):                    4.40e-53
Time:                        12:32:57   Log-Likelihood:                         -42.246
No. Observations:                  30   AIC:                                      86.49
Df Residuals:                      29   BIC:                                      87.89
Df Model:                           1                                                  
Covariance Type:            nonrobust                                                  
==================================================================================
                     coef    std err          t      P>|t|      [0.025      0.975]
----------------------------------------------------------------------------------
Hours_of_Study    10.3278      0.032    321.961      0.000      10.262      10.393
==============================================================================
Omnibus:                        4.058   Durbin-Watson:                   1.460
Prob(Omnibus):                  0.131   Jarque-Bera (JB):                2.822
Skew:                           0.586   Prob(JB):                        0.244
Kurtosis:                       2.061   Cond. No.                         1.00
==============================================================================

Notes:
[1] R² is computed without centering (uncentered) since the model does not contain a constant.
[2] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

b
Assumptions of Simple Linear Regression and How They Apply
Linearity:

The relationship between study hours and exam scores should be straight-line-like.
With a perfect R² (1.000), the data shows a strong linear trend. Looks good!
Independence of Errors:

Residuals (differences between predicted and actual values) shouldn’t be connected.
The Durbin-Watson score is 1.46, which is close to 2 (ideal). This means the residuals are fairly independent.
Homoscedasticity (Constant Error Spread):

The spread of the errors should be about the same across all predicted scores.
The statistical tests (Omnibus and Jarque-Bera) don’t raise any red flags, but it’s good to plot residuals to confirm.
Normality of Errors:

The residuals should behave like a bell curve (normally distributed).
Since the tests show no serious skew (p-values > 0.05), this assumption is also looking good.

c
Outliers can cause problems:

Mess with the slope and intercept: They can pull the line toward themselves, making predictions less accurate.
Reduce prediction quality: If the errors become too big, the model won’t work well.
Cause instability: Some points can have too much power (high leverage) and impact the entire model.

d

1) Check for residuals
2) Look for Influential Points
3) Test for normality
4) Try Cross-validation